kubectl get cep returns empty columns of policies statuses

romanspb80 opened the pull request:
This patch will set "Status N.A." for the fields INGRESS ENFORCEMENT, EGRESS ENFORCEMENT, VISIBILITY POLICY if Cilium run with --endpoint-status disabled.

Fixes: #17880

Signed-off-by: Roman Ptitcyn romanspb@yahoo.com

aditighag reviewed (commented):
Thanks! Can you add a snippet of the command output to the commit description with your changes?

aditighag reviewed (changes requested):
For the test failures - 

BPF patchcheck: Can you add a sign-off according to the guidelines - https://docs.cilium.io/en/stable/contributing/development/contributing_guide/? 
golang generate-k8s-api: 
>Please run make generate-k8s-api and submit your changes.
Merge ability: I've added a release note.

squeed commented on pkg/endpoint/endpoint_status.go:
```diff
@@ -20,6 +21,11 @@ import (
 
+const (
+	// StatusNA is value of fields in the outpu of 'kubectl get cep' in case of disabled --endpoint-status
+	StatusNA = "Status N.A."
```
suggestion: just the string "disabled".

romanspb80 replied:
> suggestion: just the string "disabled".

So it was decided when discussing the issue https://github.com/cilium/cilium/issues/17880

aanm replied:
Yes, presenting it as "disabled" would suggest the user that the policy enforcement is not enabled.

romanspb80 replied:
OK. Thanks

squeed commented on pkg/k8s/apis/cilium.io/v2/types.go:
```diff
@@ -141,6 +141,7 @@ type EndpointPolicyDirection struct {
 	Removing AllowedIdentityList `json:"removing,omitempty"`
 	// Deprecated
 	Adding AllowedIdentityList `json:"adding,omitempty"`
+	State  string              `json:"state,omitempty"`
```
Bikeshedding: `state` generally implies a machine-readable enumeration, rather than a freeform string. Perhaps this could be something like
```go
type EndpointPolicyState string

const (
    EndpointPolicyStateEnforcing EndpointPolicyState = "enforcing"
    EndpointPolicyStateNonEnforcing EndpointPolicyState = "non-enforcing"
    EndpointPolicyStateDisabled EndpointPolicyState = "disabled"
)
```

aanm replied:
Yes, agreed. An enum would probably be better suitable for this case.

romanspb80 replied:
Ok. Will do

aanm reviewed (changes requested):
As @gandro point it out, this will require a couple more changes, namely incrementing the patch value of https://github.com/cilium/cilium/blob/24a7e412b1196e829d76c22ad44613d56937d58b/pkg/k8s/apis/cilium.io/v2/register.go#L26

romanspb80 committed:
kubectl get cep returns empty columns of policies statuses

This patch set "disabled" for the fields INGRESS ENFORCEMENT, EGRESS ENFORCEMENT, VISIBILITY POLICY if Cilium run with --endpoint-status disabled.
$ kubectl get cep
NAME                        ENDPOINT ID   IDENTITY ID   INGRESS ENFORCEMENT   EGRESS ENFORCEMENT   VISIBILITY POLICY   ENDPOINT STATE   IPV4          IPV6
deathstar-f694cf746-f2tf4   1239          28002         disabled              disabled             disabled            ready            10.11.0.52    fd04::6e90
deathstar-f694cf746-gnrg6   3603          28002         disabled              disabled             disabled            ready            10.11.0.78    fd04::6dd9
tiefighter                  810           35802         disabled              disabled             disabled            ready            10.11.0.72    fd04::3ccb
xwing                       1629          58767         disabled              disabled             disabled            ready            10.11.0.136   fd04::bc93                    56            58767         Status N.A.           Status N.A.          Status N.A.         ready            10.11.0.53    fd04::d70d

Fixes: #17880

Signed-off-by: Roman Ptitcyn <romanspb@yahoo.com>
